session/objecthandling.go

```diff
@@ -292,14 +299,22 @@ func UpdateObjectHyperCubeDataAsync(sessionState *State, actionState *action.Sta
 			return errors.Errorf("object<%s> has no hypercube", gob.GenericId)
 		}
 
-		if hypercube.Size == nil {
-			return errors.Errorf("object<%s> has no hypercube size", gob.GenericId)
-		}
-
 		if err := checkHyperCubeErr(gob.GenericId, hypercube.Error); err != nil {
+			switch err.(type) {
+			case CalcEvalConditionFailedError:
+				sessionState.LogEntry.Logf(logger.WarningLevel, "object<%s>: %v", obj.ID, err)
+				if err = obj.SetHyperCubeDataPages(make([]*enigma.NxDataPage, 0), false); err != nil {
+					return errors.Wrap(err, "failed to set hypercube datapages")
```

Comment:
You could just
> return errors.Wrap(err, "failed to set hypercube datapages")

unconditionally because the line after returns a nil. In this case err is nil so the wrap would return nil as well.

Reply:
yes true, just copy pasted the lines from the different set data parts, but makes sense